src/cli/commands/snip.ts

```diff
@@ -73,6 +70,42 @@ const handler = async ({
     }
   });
 
+  // Define the handler for generating formatted snippet files.
+  bluehawk.subscribe(async (result: ParseResult) => {
+    const { source } = result;
+    if (
+      source.attributes["snippet"] === undefined ||
+      source.attributes["emphasize"] === undefined
+    ) {
+      return;
+    }
+
+    // TODO: need to make sure that "generate formatted snippets" is turned on here
+    const targetPath = path.join(destination, source.basename);
+
+    const rstHeader = ".. code-block:: ";
+    const rstEmphasizeModifier = ":emphasize-lines: ";
+    const range = source.attributes["emphasize"]["range"];
+
+    const formattedCodeblock =
+      rstHeader +
```

Comment:
Avoid using `+` to concatenate strings - instead either put each line in an array then `.join("\n")` or use template strings (backticks) with `${}` to interpolate values.